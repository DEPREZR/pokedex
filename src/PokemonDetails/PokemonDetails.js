import React, { useContext } from "react";
import { useParams } from "react-router-dom";
import useGet from "apiHelpers/hooks/useGet";
import Loader from "Loader";
import PokemonSprites from "./components/PokemonSprites";
import PokemonStats from "./components/PokemonStats";
import { Card } from "antd";

const PokemonDetailsContext = React.createContext();

export const usePokemonDetailsContext = () => useContext(PokemonDetailsContext);

const PokemonDetails = () => {
  const { pokemonId } = useParams();
  const { data: pokemon, loading } = useGet(`pokemon/${pokemonId}`);

  return (
    <Loader loading={loading}>
      {() => (
        <PokemonDetailsContext.Provider value={pokemon}>
          <Card title={pokemon.name}>
            <div className="d-flex">
              <div style={{ height: 350, width: 250, marginRight: 16 }}>
                <PokemonSprites />
              </div>
              <PokemonStats />
            </div>
          </Card>
        </PokemonDetailsContext.Provider>
      )}
    </Loader>
  );
};

export default PokemonDetails;
