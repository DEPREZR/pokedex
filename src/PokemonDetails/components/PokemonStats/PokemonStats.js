import React from "react";
import { usePokemonDetailsContext } from "PokemonDetails";
import { Card, List, Typography, Divider } from "antd";

const { Text } = Typography;

const statsPrettyNames = {
  hp: "Base HP",
  attack: "Base attack",
  defense: "Base defense",
  "special-attack": "Base special attack",
  "special-defense": "Base special defense",
  speed: "Base speed"
};

const PokemonStats = () => {
  const pokemon = usePokemonDetailsContext();

  return (
    <Card style={{ width: 348 }}>
      <List split={false}>
        <List.Item
          style={{ width: 300 }}
          className="d-flex flex-row justify-content-between"
        >
          <Text strong>Type(s)</Text>
          {pokemon.types.map(type => type?.type?.name).join(", ")}
        </List.Item>
        <List.Item
          style={{ width: 300 }}
          className="d-flex flex-row justify-content-between"
        >
          <Text strong>Weight</Text>
          {pokemon.weight}
        </List.Item>
        <Divider />
        {pokemon.stats.map(stat => (
          <List.Item
            key={stat.stat.name}
            style={{ width: 300 }}
            className="d-flex flex-row justify-content-between"
          >
            <Text strong>{statsPrettyNames[stat.stat.name]}</Text>
            {stat.base_stat}
          </List.Item>
        ))}
      </List>
    </Card>
  );
};

export default PokemonStats;
