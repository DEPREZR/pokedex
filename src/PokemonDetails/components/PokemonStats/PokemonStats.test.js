import PokemonStats from ".";
import { shallow } from "enzyme";
import { usePokemonDetailsContext } from "PokemonDetails";

jest.mock("PokemonDetails", () => {
  const originalModule = jest.requireActual("PokemonDetails");

  return {
    __esModule: true,
    ...originalModule,
    usePokemonDetailsContext: jest.fn(),
  };
});

const pokemonMock = {
  order: 35,
  name: "pikachu",
  sprites: {
    back_default:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png",
    back_female:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/female/25.png",
    back_shiny:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/25.png",
    back_shiny_female:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/female/25.png",
    front_default:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
    front_female:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/female/25.png",
    front_shiny:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/25.png",
    front_shiny_female:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/female/25.png",
    other: {
      dream_world: {
        front_default:
          "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/25.svg",
        front_female: null,
      },
      "official-artwork": {
        front_default:
          "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/25.png",
      },
    },
  },
  stats: [
    {
      base_stat: 35,
      effort: 0,
      stat: {
        name: "hp",
        url: "https://pokeapi.co/api/v2/stat/1/",
      },
    },
    {
      base_stat: 55,
      effort: 0,
      stat: {
        name: "attack",
        url: "https://pokeapi.co/api/v2/stat/2/",
      },
    },
    {
      base_stat: 40,
      effort: 0,
      stat: {
        name: "defense",
        url: "https://pokeapi.co/api/v2/stat/3/",
      },
    },
    {
      base_stat: 50,
      effort: 0,
      stat: {
        name: "special-attack",
        url: "https://pokeapi.co/api/v2/stat/4/",
      },
    },
    {
      base_stat: 50,
      effort: 0,
      stat: {
        name: "special-defense",
        url: "https://pokeapi.co/api/v2/stat/5/",
      },
    },
    {
      base_stat: 90,
      effort: 2,
      stat: {
        name: "speed",
        url: "https://pokeapi.co/api/v2/stat/6/",
      },
    },
  ],
  types: [
    {
      slot: 1,
      type: {
        name: "electric",
        url: "https://pokeapi.co/api/v2/type/13/",
      },
    },
  ],
  weight: 60,
};

describe("PokemonStats", () => {
  it("renders correctly", () => {
    usePokemonDetailsContext.mockReturnValueOnce(pokemonMock);
    const wrapper = shallow(
        <PokemonStats />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
