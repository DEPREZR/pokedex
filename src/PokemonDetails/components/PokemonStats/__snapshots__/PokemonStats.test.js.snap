// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonStats renders correctly 1`] = `
<Card
  style={
    Object {
      "width": 348,
    }
  }
>
  <List
    split={false}
  >
    <Item
      className="d-flex flex-row justify-content-between"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <Text
        strong={true}
      >
        Type(s)
      </Text>
      electric
    </Item>
    <Item
      className="d-flex flex-row justify-content-between"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <Text
        strong={true}
      >
        Weight
      </Text>
      60
    </Item>
    <Divider />
    <Item
      className="d-flex flex-row justify-content-between"
      key="hp"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <Text
        strong={true}
      >
        Base HP
      </Text>
      35
    </Item>
    <Item
      className="d-flex flex-row justify-content-between"
      key="attack"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <Text
        strong={true}
      >
        Base attack
      </Text>
      55
    </Item>
    <Item
      className="d-flex flex-row justify-content-between"
      key="defense"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <Text
        strong={true}
      >
        Base defense
      </Text>
      40
    </Item>
    <Item
      className="d-flex flex-row justify-content-between"
      key="special-attack"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <Text
        strong={true}
      >
        Base special attack
      </Text>
      50
    </Item>
    <Item
      className="d-flex flex-row justify-content-between"
      key="special-defense"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <Text
        strong={true}
      >
        Base special defense
      </Text>
      50
    </Item>
    <Item
      className="d-flex flex-row justify-content-between"
      key="speed"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <Text
        strong={true}
      >
        Base speed
      </Text>
      90
    </Item>
  </List>
</Card>
`;
